integrationtests/self/http_test.go

```diff
@@ -1042,11 +1043,18 @@ var _ = Describe("HTTP tests", func() {
 		resp, err := client.Get(fmt.Sprintf("https://localhost:%d/trailers", port))
 		Expect(err).ToNot(HaveOccurred())
 		Expect(resp.StatusCode).To(Equal(200))
-		Expect(resp.Header.Values("Trailer")).To(Equal([]string{"AtEnd1, AtEnd2", "LAST"}))
+		Expect(resp.Header.Get("Trailer")).To(Equal(""))
```

Comment:
This doesn't match Go's standard library behavior, see the screenshot in https://github.com/quic-go/quic-go/pull/4630#discussion_r1739962302.

Reply:
Actually, I think it does. Over the wire, the trailers are sent but on the client side the "Trailer" header is processed and then removed from the list of headers before being given to the caller. See here: https://github.com/golang/go/blob/2707d42966f8985a6663c93e943b9a44b9399fca/src/net/http/transfer.go#L781

Reply:
You're right, sorry for the confusion.